### PR TITLE
U11 PR2: entry contract under the merged column + a real intake-column bug the audit surfaced

### PR DESCRIPTION
**Pull request by gsxdsm**
Second small PR for **U11**. Two commits: a tests-only entry-contract pin, then a **real present-day bug fix** the audit surfaced.

## The audit you asked for, finished — no design fork

You named four surfaces as the remaining risk. All four can take a combined `intake` + `hold` column. One needed a code change; here it is.

| Surface | Verdict | Evidence |
|---|---|---|
| `isUnplannedForExecution` | Safe | PR1 (#2495) — passed unmodified; a mutation now fails exactly the merged-column test |
| Capacity hold / release | Safe | PR1 — `hold-release.ts:260` already accepts intake **or** hold |
| `start`'s column / entry contract | Safe | commit 1 — all 6 assertions passed unmodified |
| `createTask` intake wiring | **Broken today** | commit 2 — fixed, revert-proven |
| *(also found)* triage auto-discovery | Needs conversion | `triage.ts:1382` — deferred to PR3, see below |

## Commit 1 — entry contract under the merged column (tests only)

All 6 new assertions passed on the first run. **Regression floor, not evidence of a fix** — I could not make them fail and am not claiming otherwise.

They pin one real behavioral **difference** rather than asserting sameness everywhere: the merged shape answers `start` where the split shape answers `plan`, because `start` becomes the first node in that column once the columns collapse. That is equivalent *only* because `start` reaches the specification node by a single unconditional success edge — asserted, so if a node is ever inserted between them this fails instead of silently admitting an unspecified card into implementation.

Also pinned: past planning both shapes agree exactly; a card past the merged column still never resumes at a planning node (the backward drag that fires `abort-on-exit`); and a row persisted in the **deleted** `triage` column resolves to `undefined`, safe only while the executor's start-node fallback exists.

## Commit 2 — a real bug, found by the audit

The intake column was resolved **only** as a by-product of materializing workflow steps. A create supplying `enabledWorkflowSteps` without an explicit `workflowId` takes **neither** materialization branch, so `resolvedEntryColumn` stays `undefined` and `column:` falls through to the hard-coded `|| "triage"`.

Today, on Coding (Ideas), that lands the card in `triage` — **a column that workflow does not declare.** Created straight into a phantom lane. Measured: the new test fails `expected 'triage' to be 'ideas'` against unmodified sources.

**Why it blocks U11.** Once `triage` leaves the coding IRs this stops being an Ideas edge case and becomes the default workflow's behavior for every create down this path: the card lands in an undeclared column **and** — because `isIntakeColumn` keys on the same `"triage"` literal — gets `generateSpecifiedPrompt` instead of the bootstrap seed. Triage admits a card for planning only when its `PROMPT.md` reads as a seed, so a placeholder spec is classified "already planned" and never planned. The card sits in Planning forever with no log line in any lane — **FN-8587's exact failure mode, promoted from one edge case to every new card.**

The fix resolves the intake column **side-effect-free** (read the IR, ask which column carries `intake`). It deliberately does *not* call `materializeDefaultWorkflowSteps`, which would persist step rows the caller explicitly opted out of by supplying its own toggles. Unresolvable workflow returns `undefined` and each call site keeps its legacy fallback, so no path loses behavior when the IR cannot be read.

Applied to both create paths. Branch ordering preserved in both — the explicit empty-toggle case (`length === 0` hydrating back as `[]`) still runs, now nested rather than sequential.

**Revert check:** with `task-creation.ts` reverted, *"lands a Coding (Ideas) task in ideas even when enabledWorkflowSteps is supplied"* fails `expected 'triage' to be 'ideas'`. The companion bootstrap-`PROMPT.md` assertion passes either way today — it is correct **by accident of the `"triage"` literal** — and is kept precisely because that accident disappears with U11.

## Verification

37 tests green across the three intake/create suites; 119 across the entry-contract, merged-column and lifecycle suites; `pnpm test:gate` green (307 + 10 + 71); lint and core typecheck clean. Changeset added.

## Deferred to PR3, with the line numbers

`discoverReadyPlanningTasks` has two hardcoded branches:

```ts
(t) => t.column === "triage" && isTaskStillInPlanningStage(t)   // triage.ts:1382
(t) => t.column === "todo"   && !this.processing.has(t.id) …    // triage.ts:1389
```

Delete `triage` and branch 1 matches nothing for coding cards; branch 2 then does all the work and is **narrower** (it admits only `needs-replan` or bootstrap-stub cards). Commit 2 is what makes branch 2 sufficient — every new card now gets a real bootstrap seed. They cannot double-fire: a card is in `todo` xor `triage`.

Two adjacent sites are already merged-shape-ready: `triage.ts:3899` skips the redundant same-column move for a plan-in-place card, and `triage.ts:753`'s stale-status sweep already scans both columns.

Then the ~10-line IR change, then the migration proof.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/engine/src/__tests__/workflow-graph-entry-contract.test.ts [hunk 1]
```diff
@@ -170,3 +170,135 @@ describe("workflow graph entry contract — the executor honors it", () => {
...
+        { from: "start", to: "plan", condition: "success" },
+        { from: "plan", to: "plan-review", condition: "success" },
+        { from: "plan-review", to: "parse", condition: "success" },
+        { from: "plan-review", to: "plan-replan", condition: "failure" },
+        { from: "plan-replan", to: "plan-review", condition: "success", kind: "rework" },
+        { from: "parse", to: "review", condition: "success" },
+        { from: "review", to: "merge-gate", condition: "success" },
+        { from: "merge-gate", to: "end", condition: "success" },
+      ],
+    } as unknown as WorkflowIr;
```
[the remark this answers is not in the document]
Comment (gsxdsm) [reply packages/engine/src/__tests__/workflow-graph-entry-contract.test.ts]: Agreed, and rewritten rather than patched — in 5a36c5d02.

The synthetic graph is **deleted**. `mergeTodoIntoPlanning` is now the literal U11 edit expressed as a transformation — merge `triage`'s traits into `todo`, rename it "Planning", delete `triage`, repoint every node that named it — applied to the **real production IR**. The assertions track production topology by construction rather than by resemblance.

A new test guards the transformation itself: no `triage` column survives, no node still names it, `intake` and `hold` are both present, and the node **count** is unchanged (this is a column merge, not a graph edit). Without that, a broken transform would quietly make every assertion below measure the wrong thing.

While rewriting I found a related weakness neither thread raised, so flagging it rather than leaving it: two of my merged-vs-production comparisons were **tautological**. If production topology drifted, both sides of the comparison would move together and the test would stay green while the merged board silently skipped a review gate. Those now pin the **absolute** node ids (`parse`, `browser-verification`) first — the same literals the production block above pins — and keep the differential as a second assertion.

File: packages/engine/src/__tests__/workflow-graph-entry-contract.test.ts [hunk 2]
```diff
@@ -170,3 +170,135 @@ describe("workflow graph entry contract — the executor honors it", () => {
...
+  it("still resumes a card stranded in the DELETED triage column without dropping it", () => {
+    /*
+    R7 / migration case: rows persisted in `triage` outlive the column. `resolveColumnResumeNode`
+    returns undefined for a column the IR does not declare, and the executor's caller falls back to
+    the start node — so the card re-enters at the top of the pipeline rather than being stranded.
+    Pinned because "undefined" here is only safe while that fallback exists.
+    */
+    expect(resolveColumnResumeNode(mergedIr, "triage")).toBeUndefined();
+    expect(mergedIr.nodes.find((node) => node.kind === "start")?.id).toBe("start");
+  });
```
[the remark this answers is not in the document]
Comment (gsxdsm) [reply packages/engine/src/__tests__/workflow-graph-entry-contract.test.ts]: Correct, and this was the more serious of the two. Fixed in 5a36c5d02.

You identified exactly the gap: the test asserted that `resolveColumnResumeNode` returns `undefined` and that a start node exists. **Both of those hold while cards strand.** The thing that actually rescues a card stored in the deleted column is the `?? ir.nodes.find(kind === "start")` fallback inside `run()`, and neither assertion touched it.

The case now drives `executor.run()` on a card stored in the deleted `triage` column against the merged production IR, and asserts it really traverses — a non-empty trace beginning at the graph's start node. An empty trace *is* the stranding.

**Verified against the bar rather than assumed.** Deleting the fallback:

```diff
- : resolveColumnResumeNode(ir, task.column) ?? ir.nodes.find((node) => node.kind === "start");
+ : resolveColumnResumeNode(ir, task.column);
```

makes exactly this test fail with `WorkflowIrError: Workflow IR missing start node`, while the other 14 stay green. **The previous version of the test stayed green through that same deletion** — which is the whole of your point, and why it was worth rewriting rather than adding an assertion.

[the remark this answers is not in the document]
Comment (gsxdsm) [reply packages/engine/src/__tests__/workflow-graph-entry-contract.test.ts]: Correct, and this was the more serious of the two. Fixed in 5a36c5d02.

You identified exactly the gap: the test asserted that `resolveColumnResumeNode` returns `undefined` and that a start node exists. **Both of those hold while cards strand.** The thing that actually rescues a card stored in the deleted column is the `?? ir.nodes.find(kind === "start")` fallback inside `run()`, and neither assertion touched it.

The case now drives `executor.run()` on a card stored in the deleted `triage` column against the merged production IR, and asserts it really traverses — a non-empty trace beginning at the graph's start node. An empty trace *is* the stranding.

**Verified against the bar rather than assumed.** Deleting the fallback:

```diff
- : resolveColumnResumeNode(ir, task.column) ?? ir.nodes.find((node) => node.kind === "start");
+ : resolveColumnResumeNode(ir, task.column);
```

makes exactly this test fail with `WorkflowIrError: Workflow IR missing start node`, while the other 14 stay green. **The previous version of the test stayed green through that same deletion** — which is the whole of your point, and why it was worth rewriting rather than adding an assertion.
